March into Mapness - Traveller's Rest (Space structure)

LordVollkorn opened the pull request:
## About The Pull Request

I thought it would be fun to learn mapping with [this guide](https://hackmd.io/QBaz0CaTRhOp7pEnUlXnzQ), and perhaps along the way I could make a bit of content that I would like to see in space some day.

I wanted to create an only recently abandoned space structure that is relatively small but could be used as a pitstop for space-travelling spessmen to rest and recharge, perhaps even be used as a second base for further exploration.

The structure starts unpowered, but after you break in, you can turn on the APC (starts unlocked) and set the solar panels to generate power. With all it's amenities, it's similar to the "Space Bunker" (deepstorage.dmm), but of course much smaller. There is not much in terms of loot, just a random toolbox, night vision goggles and a better capacitator for the charger.

With all the new space ruins upcoming, perhaps its not the worst idea to have a place like this.

![grafik](https://user-images.githubusercontent.com/66637090/224802255-190ef0d2-251a-4456-9db9-5bfc9e9d689b.png)
![grafik](https://user-images.githubusercontent.com/66637090/224801640-c39e41c3-e98c-4619-90e1-8ebe406e86a3.png)

_Since this is my first map, I might have missed some things.
I am a little bit torn between "it does not look abandonded enough" and "it should look nice because its a resting point"._

### Mapping March

Ckey to recieve rewards: LordVollkorn

## Changelog
:cl: LordVollkorn / Gage Gaebler
add: Added Traveler's Rest, a tiny space structure for space travellers to relax and to prepare for the explorations ahead.
/:cl:

LordVollkorn committed:
Traveller's Rest (Map)

A new random space structure, relatively small, but offers a few amenities.

ZephyrTFA commented:
What's the point of the four 1x1's with windoors?

ZephyrTFA commented:
wrong button

Mothblocks commented on code/datums/ruins/space.dm:
```diff
@@ -350,3 +350,9 @@
 	name = "Alternate Sciences Research Center"
 	description = "When you're messing with the occult, who knows what you're going to get?"
+
+/datum/map_template/ruin/space/travellersrest
```
`travellers_rest`

LordVollkorn replied:
just in this one spot or for all instances of this word?

Mothblocks replied:
All preferably

LordVollkorn replied:
All changed

LordVollkorn commented:
> What's the point of the four 1x1's with windoors?

I really liked the light arrangement and the lights were originally encased in a full glass tile, but then I noticed lights were breaking and they couldnt be reached without deconstruction.
So I figured with windoors I could keep the looks, still reach the lights, and perhaps gain the one tile as storage space.
(Might sound strange but I couldnt find a way to make the lights reachable and make it look nice)

LordVollkorn committed:
Merge branch 'master' into SpaceRuin

LemonInTheDark commented:
what does it look like with starlight? (uncomment the STARLIGHT config in game_options.txt)

LordVollkorn committed:
Adjusted naming

LordVollkorn committed:
Merge branch 'SpaceRuin' of https://github.com/LordVollkorn/ChemistryStation into SpaceRuin

LordVollkorn commented:
> what does it look like with starlight? (uncomment the STARLIGHT config in game_options.txt)

Unsure if there is a visual difference from the screenshots from the description above

![grafik](https://user-images.githubusercontent.com/66637090/224455029-c629da3d-2152-4089-9fc8-b413ab2d977f.png)

![grafik](https://user-images.githubusercontent.com/66637090/224455047-a035632d-b719-4999-a7a6-2468f0e2bc33.png)
![grafik](https://user-images.githubusercontent.com/66637090/224455059-4bc12382-7bdc-4206-b1e1-ef1b213930df.png)

LordVollkorn committed:
Merge branch 'master' into SpaceRuin

Mothblocks commented:
You don't have to click the update branch button, it is not an error

Mothblocks commented:
![image](https://user-images.githubusercontent.com/35135081/224456232-8eb1c3a7-1cff-40d2-85ed-224d220cb8b2.png)

LordVollkorn committed:
Replaced window spawners with windows and removing unneccessary cable

Replaced window spawners with windows and removing unneccessary cable

LordVollkorn committed:
Also removed the grille, wondering why that was an issue,

Also removed the grille, wondering why that was an issue,

MMMiracles commented:
I like the general layout and aesthetic, especially for a first-time map. I only really have some suggestions apart from two minor mapping issues.

Issues:
- Circled solar arrays have doubled up panels.
![image](https://user-images.githubusercontent.com/9276171/224460053-4a038b56-e4ba-4a8d-9b70-78e8731c2c38.png)
- The area next to the grinder is using the ruin's area instead of the area passthrough.

Suggestions:
- Maybe add some lattices outside with some basic debris protection (i.e grilles with girders under them)? I don't believe space debris is actually an issue outside of the station z-level but thematically it seems weird to have something so fragile with zero debris catchers.
- Maybe replace the large lights with fewer/small bulb ones? That APC is probably gonna be chugging hard with 8 large lights on top of all the other existing equipment on that grid, especially since it has no SMES to store excess power and I assume that APC is on a standard cell by default.
- I would suggest at least adding a spot for a spare oxygen tank and a mask. I would of also said even a super simple atmos piping (an air tank for distro and waste pumping into space) but the ruin itself is pretty small and having to add one would mess up the current layout.

Otherwise looks cool. Would get drunk in if I fell out the escape shuttle and flew into this.

private-tristan commented:
Do holopads work off Zlevel? If so it could be a unique way for space explorers to contact the station.

VladinXXV commented:
toilet right next to the window. funny. bold?
![image](https://user-images.githubusercontent.com/44104681/224465938-c831f877-d010-408c-9401-321cbc15ef22.png)

coiax commented:
My main concerns are about long term thematic/mechanical viability:

APCs generally have small batteries, are the solar panels enough to sustain the base at full power?
With no air supply, there'd be an upper limit to how long you could stay in this place. I doubt you'd reach it in normal gameplay though.

Could add an air cannister or a portable scrubber to acknowledge the atmospheric situation, an adhoc atmos setup.

I suppose thematically you could say the potted plants consume CO2, but also they're like immortal unchangable plants that require no water.

(Also, I'll get yelled at by the mapping crew, but I do not like tiny fans to "keep the vaccum" out. But I don't think we really have a proper airlock chamber setup that's easily usable, so not something that can be solved quickly)

---

But also none of these concerns are enough to stop this cute little ruin being added. 🙇

LordVollkorn commented:
@MMMiracles Wow, thanks! I didn't even notice the solar arrays were double; Is there a way of checking these things without right-clicking the tiles?
I have fixed the arrays and the space tiles accordingly 👍

About the lattice: I placed some around the structure before, but that made it a bit too big and the view out the window wasn't so nice. If possible I would like to not use the lattice at all (and like you mentioned, I also think that space debris is not generated in deep space)

Good point with the lights and the power: Before i made this PR, I actually used an SMES but I noticed that with the current setup it was not neccessary; The power levels always stay at good levels, even with all lights burning and some machinery usage.

I don't know if its possible to integrate an atmos system in such a small space, but the only related thing it has is a tiny fan in the outer airlock. The tank and internals are a good idea though; I added some to the entrace area and moved the pAI to an empty table.
![grafik](https://user-images.githubusercontent.com/66637090/224482878-b26e6389-3df1-4835-a283-2f5b7c7bd44d.png)

LordVollkorn commented:
> Do holopads work off Zlevel? If so it could be a unique way for space explorers to contact the station.
@private-tristan 
Good question, I originally added it because I wanted to have a pre-recorded hologram message of the original owner of the structure, but in the end I just left it in for the aesthetics.
Might need to pick that original idea back up, if I can find some example I can use from another map

LordVollkorn commented:
@coiax 
Good points! I have tested the structure for 20-30 minutes in a round and as far as I can tell, the APC with the solar panels attached can handle things pretty good. There is a tiny fan in the outer airlock to prevent air from escaping, but only because small mistakes (like dragging a big object behind you) could drain the structure really quickly (and I dont have enough space for an atmos setup).
 I added an O2 tank and a mask to the entrance area now, just to make sure!

I didn't give too much thought to the plants, they might be made of plastic :D

LordVollkorn committed:
Feedback Fix #1

Removed double solar arrays, fixed one outside tile to be space instead of ruin, added o2 tank and mask.

(Also replaced the garbage decal because its not cleanable with space cleaner)

LordVollkorn committed:
Holodisk, decals & parts

I figured out how to make holodisks and added one for flavor. Then I remembered decals (and I love the decal printer in game) and I couldn't stop myself.
Also remembered that the charger requires another capacitator and a power cell.

LordVollkorn commented:
Up-to-date screenshot with all changes mentioned before
![grafik](https://user-images.githubusercontent.com/66637090/224488772-cb436cbe-77d2-4fb7-bc23-9f6c2452ef3b.png)

LordVollkorn committed:
Changed holodisk preset image (runtime fix?)

Switched the hologram entity from atmos/mod to a simple atmos person. The mod helmet caused a runtime for some reason.

LordVollkorn committed:
Merge branch 'master' into SpaceRuin

newfren commented:
a booze dispenser without accompanying soda dispenser really limits the drinks that can actually be made with it

LordVollkorn commented:
> a booze dispenser without accompanying soda dispenser really limits the drinks that can actually be made with it
@newfren 
I might be able squeeze one in somehow, let me see what I can do

ArthuroPep commented:
I want to go there with a communication relay then spend the remainder of the shift talking on the radio and rping as Esker from Outer Wilds, greeting travelers among the way

LordVollkorn commented:
> I want to go there with a communication relay then spend the remainder of the shift talking on the radio and rping as Esker from Outer Wilds, greeting travelers among the way

@ArthuroPep
I like this idea, and I think those 4 compartments where the lights are in can be used to house the machines

LordVollkorn commented:
I managed to squeeze the last few pixels out of the bar area
![grafik](https://user-images.githubusercontent.com/66637090/224514936-f8efed88-b718-428b-be6e-e999c0230ede.png)

And since it was easy to add, I also included a light switch and a newscaster
![grafik](https://user-images.githubusercontent.com/66637090/224514946-d2cdb14d-e20f-4a33-a6fd-6b59dc2e012a.png)

LordVollkorn commented:
![grafik](https://user-images.githubusercontent.com/66637090/224515018-ab34e60a-07b8-4600-afed-697c541fcfd4.png)

LordVollkorn committed:
Feedback Fix #2

Bar got soda dispenser, sink, compartment. Also added light switch and newscaster

Mothblocks commented on code/datums/holocall.dm:
```diff
@@ -476,3 +476,25 @@
+	desc = "A holodisk containing a small memo from the previous owner, addressed to someone else."
+	preset_image_type = /datum/preset_holoimage/engineer/atmos
+	preset_record_text = {"
+	NAME Space Adventurer
+	SOUND PING
+	DELAY 20
+	SAY Hey, I left you this message for when you come back.
+	DELAY 50
+	SAY I picked up an emergency signal from a freighter and I'm going there to search for some goodies.
+	DELAY 50
+	SAY You can crash here if you need to, but make sure to check the anchor cables before you leave.
+	DELAY 50
+	SAY If you don't, this thing might drift off into space.
+	DELAY 50
+	SAY Then some weirdo could find it and potentially claim it as their own.
+	DELAY 50
+	SAY Anyway, gotta go, see ya!
+	DELAY 40
+	SOUND sparks"}
```
Indent these all up by one, and have `"}` on its own line (dedented one)

LordVollkorn replied:
@Mothblocks Yes sir

Mothblocks commented on code/game/area/areas/ruins/space.dm:
```diff
@@ -570,3 +570,7 @@
 	name = "\improper Clown Planet"
 	ambientsounds = list('sound/ambience/clown.ogg')
+
+// The abandonded capsule 'The Travellers Rest'
```
```suggestion
// The abandoned capsule 'The Traveller's Rest'
```

Mothblocks commented:
I also just realized it's spelled "Traveler's" and not "Traveller's" if you care enough

LordVollkorn committed:
Indentation fix

LordVollkorn committed:
Comment fix

LordVollkorn committed:
Never use WASD in the editor

LordVollkorn commented:
> I also just realized it's spelled "Traveler's" and not "Traveller's" if you care enough

@Mothblocks 
I had to google it and you are right
![grafik](https://user-images.githubusercontent.com/66637090/224515391-6375769f-3b48-4aa0-bb28-e8f85fadad1d.png)

I should be able to change that

LordVollkorn committed:
Naming changed

LordVollkorn committed:
Merge branch 'master' into SpaceRuin

Mothblocks commented:
Oh if it's a British thing then you didn't need to care (we explicitly don't have a preference between US and British english), I didn't know that was a regional thing

LordVollkorn commented:
All good, we can go with the US version, I have no objections either way 👍

LordVollkorn committed:
Added tiny atmos system

I woke up this morning and I had an idea of how to integrate an atmos system. Might be overkill, but I enjoyed squeezing the pixels

LordVollkorn committed:
Used generic solar panels instead of "fore-port solar array" ones

LordVollkorn committed:
Merge branch 'master' into SpaceRuin

LordVollkorn commented:
Tiny "atmos system"
![grafik](https://user-images.githubusercontent.com/66637090/224537302-e0d7ef4f-91ad-4a54-9f66-3487725b25b5.png)

LordVollkorn committed:
Merge branch 'master' into SpaceRuin

LordVollkorn committed:
Update travelers_rest.dmm

LordVollkorn committed:
Merge remote-tracking branch 'upstream/master' into SpaceRuin

LordVollkorn commented:
I feel pretty good about this map now, and with all the feedback and fixed issues I would say that this is ready to go.

MapDiffBot2 seems to error out, but what I see from other PR's this seems to be non-critical?

Zytolg commented:
For the Atmos System, could you reverse the tank to DIR_8 [WEST] and route the pipes down to a default on_state pump with 108 pressure? The way atmos is set up on this is going to result in your pressure tank draining to the capacity of the pipes (4500psi?) in an effort to reach pressure equilibrium and making any player-facing modifications to this ruin hazardous at best (they'll be blown away from unbranching pipes). You'll also want to set a waste loop for air on another pipe network and at least hook up a passive vent into space for it. Otherwise you'll just be storing air inside the scrubbers network until it hits that pipe psi cap and cease functioning.

LordVollkorn commented:
> For the Atmos System, could you reverse the tank to DIR_8 [WEST] and route the pipes down to a default on_state pump with 108 pressure? The way atmos is set up on this is going to result in your pressure tank draining to the capacity of the pipes (4500psi?) in an effort to reach pressure equilibrium and making any player-facing modifications to this ruin hazardous at best (they'll be blown away from unbranching pipes). You'll also want to set a waste loop for air on another pipe network and at least hook up a passive vent into space for it. Otherwise you'll just be storing air inside the scrubbers network until it hits that pipe psi cap and cease functioning.

@Zytolg
Hi, you good some good points there.
For the tank: I attached the pump like you said, I didn't even think about it until you mentioned it. But I opted for facing it south, because its easier to reach when under the table (in my opinion).
Or is there a specific reason to go west first?

I put the red pipes in layer 2 like on the normal station and I placed a passive vent outside (before I used an injector, it was under the solar panel)

LordVollkorn committed:
Placed air pump to avoid pressure buildup in the pipes, replaced outside injector with passive vent, changed red pipe layer to 2.

LordVollkorn committed:
Merge branch 'master' into SpaceRuin

coiax commented:
Much as I admire your efforts in cramming in the machinery into the cramped bar system, is it possible to replicate that setup if you were constructing it only ingame? That's what I'm concerned about with the pixel shifted microwaves and chemical setups.

(Note I'm not saying "oh you should totally modify the whole construction system to make this possible", it's more I have a natural dislike of any mapping setup that could not be constructed in-universe.

You could move the air tank to be on the outside, rather than inside, and if you put another table on the turf with the air alarm, then you could store the beakers and damp rag on the corner table. 🤔

LordVollkorn commented:
> Much as I admire your efforts in cramming in the machinery into the cramped bar system, is it possible to replicate that setup if you were constructing it only ingame? That's what I'm concerned about with the pixel shifted microwaves and chemical setups.
> 
> (Note I'm not saying "oh you should totally modify the whole construction system to make this possible", it's more I have a natural dislike of any mapping setup that could not be constructed in-universe.)
> 
> You could move the air tank to be on the outside, rather than inside, and if you put another table on the turf with the air alarm, then you could store the beakers and damp rag on the corner table. 🤔

@coiax 
I get what you mean; Basically its not possible (as far as I know) to replicate the bar or chemical setup with ingame tools. If this map was a fully fledged station, then of course I shouldn't work with pixel shifts like this.

But just in this case, for a unique, tiny space structure, I would like to keep the design as a "gimmicky tinyhome in space" if that makes sense 😁

Addust commented:
oh my god yes

VladinXXV commented:
> But just in this case, for a unique, tiny space structure, I would like to keep the design as a "gimmicky tinyhome in space" if that makes sense 😁

![image](https://user-images.githubusercontent.com/44104681/224775927-328134dc-3880-413e-9300-c11608f4953e.png)

You could de-pixel shift the microwave and move the All-In-One Grinder to the table where the box of drinking glasses and rag are. This keeps them all roughly in the same area while allowing it to be rebuilt ingame more accurately.

![image](https://user-images.githubusercontent.com/44104681/224776925-728ebb47-f0e5-48b0-b2bc-246950cfcf11.png)

Also, the tile underneath the toil is missing the texture dot things every other tile in that section has. Intentional?

VladinXXV commented:
![image](https://user-images.githubusercontent.com/44104681/224777602-d4d366cb-48be-4870-915c-a747443c69c2.png)

I only just noticed the chem setup, too. I strongly advise against this. I would move the pAI and bottle of space cleaner onto the same table as the box of beakers, remove the table they were previously on, put the reaction chamber in that tile, and de-pixel shift all machines.

LordVollkorn committed:
Removed crammed design and added grilles to the windows again

LordVollkorn committed:
Textured toilet tile

LordVollkorn commented:
@coiax @VladinXXV 
Although I liked the crammed design, I will take your feedback and make the thing in-game possible to rebuild.
I have de-pixelshifted the bar and chemistry, and there is only ever 1 machine per tile.
(Also corrected the toilet tile, I never noticed!)

LordVollkorn committed:
Merge remote-tracking branch 'upstream/master' into SpaceRuin

LordVollkorn committed:
Fixed grille issue with linters?

LordVollkorn commented:
Newest version:
![grafik](https://user-images.githubusercontent.com/66637090/224801489-9ea69843-6a92-41c0-9b02-32640bf3a327.png)
![grafik](https://user-images.githubusercontent.com/66637090/224801640-c39e41c3-e98c-4619-90e1-8ebe406e86a3.png)

LordVollkorn committed:
Also unshifted air tank and moved space cleaner so the table is less cramped

san7890 commented:
you must add this to https://github.com/tgstation/tgstation/blob/master/config/spaceruinblacklist.txt

LordVollkorn committed:
Added to spaceruinblacklist.txt

LordVollkorn commented:
> you must add this to https://github.com/tgstation/tgstation/blob/master/config/spaceruinblacklist.txt

@san7890 Yessir, added the line

LordVollkorn committed:
Merge remote-tracking branch 'upstream/master' into SpaceRuin

LordVollkorn committed:
Merge remote-tracking branch 'upstream/master' into SpaceRuin

san7890 reviewed (commented):
i do not believe that the walls to this structure have to be reinforced... there's not really much reason for a space ruin to be that secure? if you think there's value to it feel free to keep it but I don't really see much of any

LordVollkorn committed:
Replaced reinforced walls with normal walls

LordVollkorn committed:
Merge remote-tracking branch 'upstream/master' into SpaceRuin

LordVollkorn commented:
> i do not believe that the walls to this structure have to be reinforced... there's not really much reason for a space ruin to be that secure? if you think there's value to it feel free to keep it but I don't really see much of any

@san7890 I have taken your suggestion and replaced the walls; You were right: There was no particular reason for the walls to be reinforced, other than the looks. But for me normal walls work absolutely fine as well.

LordVollkorn committed:
Merge remote-tracking branch 'upstream/master' into SpaceRuin

Rex9001 commented:
Cool ruin! We 100% need more of these rest type areas

LordVollkorn committed:
Merge remote-tracking branch 'upstream/master' into SpaceRuin
